docs/02-app/02-api-reference/01-directives/use-cache.mdx

```diff
@@ -226,13 +226,13 @@ This inline cache profile will only be applied to the function or file it was cr
 
 ### Nested usage of `use cache` and `cacheLife`
 
-When defining multiple caching behaviors, in the same route or component tree, if the inner caches specify their own `cacheLife` profile, the outer cache will respect the shortest cache duration among them. **This applies only if the outer cache does not have its own explicit `cacheLife` profile defined.**
+When defining multiple caching behaviors in the same route or component tree, if the inner caches specify their own `cacheLife` profile, the outer cache will respect the shortest cache duration among them. **This applies only if the outer cache does not have its own explicit `cacheLife` profile defined.**
 
 **Decision hierarchy for cache boundaries:**
 
-1. Next.js will use the shortest cache profile found within the whole `use cache` boundary, excluding inner `use cache`directives.
-2. If no cache profile exists then the shortest profile times from all inner `use cache` calls applies to this `use cache`. If there are no inner `use cache`'s then the default is used
-3. Inner caches at two levels deep, do not affect the outer cache since they have already provided their duration to their parent.
+1. Next.js will use the shortest cache profile found within the whole `use cache` boundary, excluding inner `use cache` directives.
+2. If no cache profile exists, then the shortest profile times from all inner `use cache` calls applies to this `use cache`. If there are no inner `use cache`'s then the default is used
+3. Inner caches at two levels deep do not affect the outer cache since they have already provided their duration to their parent.
```

Comment:
I have to say I still don't have a clear understanding of the nested behavior. There is an example below but it doesn't actually show what's going to happen. The way I'm reading it now, I assume that `ChildComponent` will be days-stale in the context of `ParentComponent` (due to the outer boundary, rule #1) but would be hours-stale if it was rendered on its own.  Is that the expectation? This should ideally be spelled out just below the example — the code comment seems to contradict that.

Reply:
The constraints I care about as an app author:

- When I give a cache lifetime to the child, do I have a *guarantee* it will update at least as often? Even if that forces everything above to invalidate?
- Or does the outer scope "win" and outer scope can "freeze" inner scopes that would've liked to revalidate more often?
- When I want to avoid this conflict, how do I break out of it? (I assume via the `children` composition trick instead of nested scopes?)

It would be good to write this part from this perspective. The concerns that need to be addressed is (1) is my stuff ever going to be more stale than locally expected, (2) is my stuff going to revalidate more often than locally expected, (3) how to resolve conflicts between expectations if there's a clear "hole" that can be cut.

Reply:
Thank you for the feedback Dan, I'll address this and other community questions in a follow-up PR. 🙏🏼